plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -80,6 +85,27 @@ func defaultPredicates() sets.String {
 		),
 		// Fit is determined by the presence of the Host parameter and a string match
 		factory.RegisterFitPredicate("HostName", predicates.PodFitsHost),
+
+		// Fit is determined by whether or not there would be too many AWS EBS volumes attached to the node
+		factory.RegisterFitPredicateFactory(
+			"MaxEBSVolumeCount",
+			func(args factory.PluginFactoryArgs) algorithm.FitPredicate {
+				// According to Amazon, attaching more than 40 volumes can cause boot failures
+				// TODO: allow for generically parameterized scheduler predicates, because this is a bit ugly
+				var maxEBSVols uint = 39
+				if rawMaxEBSVols := os.Getenv("KUBE_MAX_EBS_VOLS"); rawMaxEBSVols != "" {
```

Comment:
I don't know how I feel about the env-var overload to set, as it's an anomaly.  Is there actually a legit use case where we would want to mock this?